src/main/java/bio/terra/workspace/service/workspace/model/Workspace.java

```diff
@@ -25,23 +67,123 @@
+        .append(workspaceId, workspace.workspaceId)
+        .append(displayName, workspace.displayName)
+        .append(description, workspace.description)
+        .append(spendProfileId, workspace.spendProfileId)
+        .append(properties, workspace.properties)
+        .append(workspaceStage, workspace.workspaceStage)
+        .append(gcpCloudContext, workspace.gcpCloudContext)
+        .isEquals();
+  }
+
+  @Override
+  public int hashCode() {
+    return new HashCodeBuilder(17, 37)
+        .append(workspaceId)
+        .append(displayName)
+        .append(description)
+        .append(spendProfileId)
+        .append(properties)
+        .append(workspaceStage)
+        .append(gcpCloudContext)
+        .toHashCode();
+  }
 
   public static Builder builder() {
-    return new AutoValue_Workspace.Builder();
+    return new Builder();
   }
 
-  @AutoValue.Builder
-  public abstract static class Builder {
-    public abstract Builder workspaceId(UUID value);
+  @JsonPOJOBuilder(withPrefix = "")
+  public static class Builder {
+    private UUID workspaceId;
+    private String displayName;
+    private String description;
+    private SpendProfileId spendProfileId;
+    private Map<String, String> properties;
+    private WorkspaceStage workspaceStage;
+    private GcpCloudContext gcpCloudContext;
+
+    public Builder workspaceId(UUID workspaceId) {
+      this.workspaceId = workspaceId;
+      return this;
+    }
+
+    public Builder displayName(String displayName) {
+      this.displayName = displayName;
+      return this;
+    }
+
+    public Builder description(String description) {
+      this.description = description;
+      return this;
+    }
+
+    public Builder spendProfileId(SpendProfileId spendProfileId) {
+      this.spendProfileId = spendProfileId;
+      return this;
+    }
+
+    public Builder properties(Map<String, String> properties) {
+      this.properties = properties;
+      return this;
+    }
 
-    public abstract Builder spendProfileId(Optional<SpendProfileId> value);
+    public Builder workspaceStage(WorkspaceStage workspaceStage) {
+      this.workspaceStage = workspaceStage;
+      return this;
+    }
 
-    public abstract Builder workspaceStage(WorkspaceStage value);
+    public Builder gcpCloudContext(GcpCloudContext gcpCloudContext) {
+      this.gcpCloudContext = gcpCloudContext;
+      return this;
+    }
 
-    public abstract Workspace build();
+    public Workspace build() {
+      // Always have a map, even if it is empty
+      if (properties == null) {
+        properties = new HashMap<>();
+      }
+      if (workspaceId == null || workspaceStage == null) {
+        throw new MissingRequiredFieldsException("Workspace requires id and stage");
+      }
+      return new Workspace(
+          workspaceId,
+          displayName,
+          description,
+          spendProfileId,
+          properties,
+          workspaceStage,
+          gcpCloudContext);
+    }
+    ;
```

Comment:
What's this semicolon for?

Reply:
Sometimes a semicolon just needs to stop always trying to do something and just practice being :)